src/vcCamera.cpp

```diff
@@ -126,22 +126,39 @@ void vcCamera_UpdateMatrices(vcCamera *pCamera, const vcCameraSettings &settings
   pCamera->matrices.projectionNear = udDouble4x4::perspectiveZO(fov, aspect, 0.5f, 10000.f);
 #endif
 
+  udDouble4x4 projectionPerspUD = udDouble4x4::perspectiveZO(fov, aspect, zNear, zFar);
+  udDouble4x4 projectionOrthoUD = udDouble4x4::orthoZO(-settings.orthographicSize * aspect, settings.orthographicSize * aspect, -settings.orthographicSize, settings.orthographicSize, vcSL_CameraOrthoNearFarPlane.x, vcSL_CameraOrthoNearFarPlane.y);
+#if GRAPHICS_API_OPENGL
+  udDouble4x4 projectionPersp = udDouble4x4::perspectiveNO(fov, aspect, zNear, zFar);
+  udDouble4x4 projectionOrtho = udDouble4x4::orthoNO(-settings.orthographicSize * aspect, settings.orthographicSize * aspect, -settings.orthographicSize, settings.orthographicSize, vcSL_CameraOrthoNearFarPlane.x, vcSL_CameraOrthoNearFarPlane.y);
+#endif
+
   switch (settings.cameraMode)
   {
   case vcCM_OrthoMap:
-    pCamera->matrices.projectionUD = udDouble4x4::orthoZO(-settings.orthographicSize * aspect, settings.orthographicSize * aspect, -settings.orthographicSize, settings.orthographicSize, vcSL_CameraOrthoNearFarPlane.x, vcSL_CameraOrthoNearFarPlane.y);
+    pCamera->matrices.projectionUD = projectionOrthoUD;
 #if GRAPHICS_API_OPENGL
-    pCamera->matrices.projection = udDouble4x4::orthoNO(-settings.orthographicSize * aspect, settings.orthographicSize * aspect, -settings.orthographicSize, settings.orthographicSize, vcSL_CameraOrthoNearFarPlane.x, vcSL_CameraOrthoNearFarPlane.y);
+    pCamera->matrices.projection = projectionOrtho;
 #endif
     break;
-  case vcCM_FreeRoam: // fall through
+  case vcCM_FreeRoam:
+    if (pCamInput->transitioningToMapMode && pCamInput->progress > 0.15)
+    {
+      //Switch to ortho projection soon after camera starts rotating downward, hence the progress threshold 0.15.
+      pCamera->matrices.projectionUD = projectionOrthoUD;
+#if GRAPHICS_API_OPENGL
+      pCamera->matrices.projection = projectionOrtho;
+#endif
+      break;
```

Comment:
This break is causing issues- if you intended this to always break you'll need to put it outside the brace below

Reply:
It's intentionally only break here when the `if` condition holds.  Otherwise this case becomes 'fall through' as how it was before. 
What issues is it causing?

Reply:
Yeah, we have strict compilation settings on so we don't get unexpected results later.